### PR TITLE
Add a yaml for snow DA testing.

### PR DESCRIPTION
**Pull request by aerorahul**
# Description
This PR:
- adds a new test yaml C96_atmsnowDA.yaml for 3DVar atmosphere with GSI and Land (Snow) DA with JEDI
- moves a few yamls from `platforms/` to `yamls/`
- adds ability to overwrite a previously created experiment as an addition to user input.

Part of #1821 

# Type of change
- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Review by TerrenceMcGuinness-NOAA — APPROVED** [review]
The over ride feature will be helpful for developers when we move to Jenkins process and and we can make changes without rebuilding.

File: ci/cases/pr/C96_atmsnowDA.yaml
Comment (CoryMartin-NOAA) [inline ci/cases/pr/C96_atmsnowDA.yaml:15]: Do we have obs for the GFS cyc? (I forgot to check)

**Review by CoryMartin-NOAA — APPROVED** [review]

File: ci/cases/pr/C96_atmsnowDA.yaml
Comment (aerorahul) [reply ci/cases/pr/C96_atmsnowDA.yaml:15]: great question @CoryMartin-NOAA 
I can disable it for now.  
On the same point, we are not running with `gfs_cyc=4` to check if the snow DA actually runs in the early cycle either (`RUN=gfs`)

Comment (jiaruidong2017) [reply ci/cases/pr/C96_atmsnowDA.yaml:15]: I didn't try to run the GFS cycles previously.

Comment (aerorahul) [reply ci/cases/pr/C96_atmsnowDA.yaml:15]: Thanks @jiaruidong2017 
I have disabled early cycle in this test.

**Comment by aerorahul** [comment]
Once @jiaruidong2017 has verified the configuration for this test is as expected, we will:
- synchronize the observations and initial conditions to Orion
- merge this PR
- run the test on via the CI

**Comment by jiaruidong2017** [comment]
@aerorahul Please don't forget to use the branch `feature/landda_ci` from GDASApp for the CI test, because one of the GTS snow data from MADIS are not available operationally on the test time. We had to prepare these observations separately.

**Comment by CoryMartin-NOAA** [comment]
@jiaruidong2017 let's get your other PR in GDASApp and update the hash as part of this PR.

**Comment by jiaruidong2017** [comment]
@CoryMartin-NOAA The `feature/order_filters` branch in GDASApp PR (https://github.com/NOAA-EMC/GDASApp/pull/790) didn't handle this. 

The new PR (https://github.com/NOAA-EMC/GDASApp/pull/839) can totally address it. @CoryMartin-NOAA Would you please review it? Thanks.

**Comment by jiaruidong2017** [comment]
I compared the config.base, and some differences are listed below: 

      Your configuration                                     My configuration
```
1)                                                       CCPP_SUITE="FV3_GFS_v17_p8"
2)   gfs_cyc=1                                           gfs_cyc=0
3)   DOIAU="YES"                                         DOIAU="NO"
```
I am not sure these differences matter.

**Review by jiaruidong2017 — APPROVED** [review]

**Comment by CoryMartin-NOAA** [comment]
We do need to account for IAU. It is up to @aerorahul if we should start with non-IAU CI and fix that "soon" or wait until we have IAU support in atmospheric DA with the snow DA.

**Comment by aerorahul** [comment]
> I compared the config.base, and some differences are listed below:
> 
> ```
>   Your configuration                                     My configuration
> ```
> 
> ```
> 1)                                                       CCPP_SUITE="FV3_GFS_v17_p8"
> 2)   gfs_cyc=1                                           gfs_cyc=0
> 3)   DOIAU="YES"                                         DOIAU="NO"
> ```
> 
> I am not sure these differences matter.

I addressed the `IAU` option and tested the `gfs_cyc=0`

I don't think I can address the `CCPP_SUITE`question.  We are defining that differently.  The variable is in `config.ufs`.  
The model is run with `CCPP_SUITE=FV3_GFS_v17_p8_ugwpv1` when running in `APP=ATM|ATMA` mode, and `CCPP_SUITE=FV3_GFS_v17_coupled_p8_ugwpv1` for coupled modes.

**Comment by aerorahul** [comment]
> We do need to account for IAU. It is up to @aerorahul if we should start with non-IAU CI and fix that "soon" or wait until we have IAU support in atmospheric DA with the snow DA.

@CoryMartin-NOAA 
I would rather have a test with `DOIAU=NO` now, rather than wait for a test till `DOIAU=YES` capability is developed.

**Review by WalterKolczynski-NOAA — COMMENTED** [review]
Looks good so far, but has conflicts to resolve and is still marked as a draft.

**Comment by aerorahul** [comment]
Conflicts have been resolved.
I believe we are waiting for an updated hash of GDASapp to be able to enable this test.

**Comment by aerorahul** [comment]
@CoryMartin-NOAA @jiaruidong2017 
I have updated this branch to include the GDASApp from this AM that includes the needed changes for ensuring this test will run and succeed.
Please confirm and we will run this test as part of this PR review.

**Comment by jiaruidong2017** [comment]
@aerorahul I checked the GDASApp, and the commit hash `d0963ed` in `feature/stable-nightly` included all my changes and could be used for this snow DA test.

**Comment by CoryMartin-NOAA** [comment]
@jiaruidong2017 what about the hash that is now checked out in global-workflow? does it include the necessary changes (it should)

**Comment by jiaruidong2017** [comment]
@CoryMartin-NOAA I will clone a new global-workflow and check out whether it is the recent version for the GDASApp.

**Comment by jiaruidong2017** [comment]
@CoryMartin-NOAA @aerorahul I confirmed that the current GDASApp hash `5056169` being checked out in global-workflow include all the necessary changes.

File: workflow/setup_expt.py
Comment (CoryMartin-NOAA) [inline workflow/setup_expt.py:572]: neat, I didn't know this trick.

File: ci/cases/pr/C96_atmsnowDA.yaml
Comment (CoryMartin-NOAA) [inline ci/cases/pr/C96_atmsnowDA.yaml]: I think @jiaruidong2017 had @KateFriedman-NOAA add in gfs dumps for the test. Up to you to test before merging or in a later PR when we do IAU we can add in gfs_cyc 1

**Review by CoryMartin-NOAA — APPROVED** [review]
Just the comment about the gfs_cyc, take it or leave it for later.

**Review by jiaruidong2017 — APPROVED** [review]

File: ci/cases/pr/C96_atmsnowDA.yaml
Comment (jiaruidong2017) [reply ci/cases/pr/C96_atmsnowDA.yaml]: Yes, it is ready for the gfs_cyc=1 test.

**Comment by aerorahul** [comment]
alright then. Hera tests have been fired off now.

File: ci/cases/pr/C96_atmsnowDA.yaml
Comment (aerorahul) [reply ci/cases/pr/C96_atmsnowDA.yaml]: Yes.  @KateFriedman-NOAA has added the obs for `gfs_cyc=1`.

**Comment by aerorahul** [comment]
> Automated global-workflow Testing Results:
> 
> ```
> Machine: Hera
> Start: Tue Jan 16 20:45:45 UTC 2024 on hfe05
> ---------------------------------------------------
> Build: Completed at 01/16/24 09:35:56 PM
> *** Failed *** to create experiment: C48_ATM_6f4bda1e on Hera
> 
> 2024-01-16 21:36:09,462 - INFO     - root        : BEGIN: __main__.input_args
> 2024-01-16 21:36:10,160 - DEBUG    - root        : (  )
> 2024-01-16 21:36:10,168 - INFO     - root        :   END: __main__.input_args
> 2024-01-16 21:36:10,168 - DEBUG    - root        :  returning: Namespace(yaml=PosixPath('/scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/2199/global-workflow/ci/cases/pr/C48_ATM.yaml'), overwrite=False)
> 2024-01-16 21:36:10,199 - INFO     - root        : Call: setup_expt.main()
> 2024-01-16 21:36:10,200 - DEBUG    - root        : setup_expt.py gfs forecast-only --pslot C48_ATM_6f4bda1e --app ATM --resdetatmos 48 --comroot /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/2199/RUNTESTS/COMROOT --expdir /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/2199/RUNTESTS/EXPDIR --idate 2021032312 --edate 2021032312 --yaml /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/2199/global-workflow/ci/cases/yamls/gfs_defaults_ci.yaml
> Traceback (most recent call last):
>   File "/scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/2199/global-workflow/workflow/create_experiment.py", line 99, in <module>
>     setup_expt.main(setup_expt_args)
>   File "/scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/2199/global-workflow/workflow/setup_expt.py", line 560, in main
>     create_comrot = query_and_clean(comrot, force_clean=user_inputs.overwrite)
> NameError: name 'comrot' is not defined
> ts_ci.yaml
> ```

This is my fault as I failed to updated the newly added yaml with the updated arguments to `setup_expt.py`

File: ci/cases/pr/C96_atmsnowDA.yaml
Comment (aerorahul) [reply ci/cases/pr/C96_atmsnowDA.yaml:15]: I re-enabled early cycle  as the obs have been added.

**Comment by jiaruidong2017** [comment]
@aerorahul I found `--idate 2021032312 --edate 2021032312` from your setup command. Should it be `--idate 2021112012 --edate 2021112100`?

**Comment by aerorahul** [comment]
> @aerorahul I found `--idate 2021032312 --edate 2021032312` from your setup command. Should it be `--idate 2021112012 --edate 2021112100`?

Not sure what you are looking at, but [C96_atmsnowDA test](https://github.com/NOAA-EMC/global-workflow/pull/2199/files#diff-6e84bed8e29d6cb9a8818a5ea0b2445d084eb4eb91ed0f87ec9eac94a81281b6) is setting the dates for which ICs were provided.

**Comment by jiaruidong2017** [comment]
@aerorahul Please use the `jiaruidong2017:feature/ens_landda` from PR https://github.com/NOAA-EMC/global-workflow/pull/2033 for your `snowtest`, because we re-organize the global-workflow jobs by moving the IMS snow process from the preplandobs job to the landanl/landensanl jobs.

**Comment by CoryMartin-NOAA** [comment]
@jiaruidong2017 can you cherry pick the relevant changes for the deterministic from your branch into a PR into this branch?

**Comment by jiaruidong2017** [comment]
@CoryMartin-NOAA Great suggestions. The relevant files are as below: 
```
jobs/rocoto/landanl.sh
parm/config/gfs/config.landanl
parm/config/gfs/config.preplandobs
scripts/exglobal_prep_land_obs.py
ush/python/pygfs/task/land_analysis.py
```
I will grab these files and create a PR into this PR.

**Comment by CoryMartin-NOAA** [comment]
I think I have a solution that will be both forward looking for helping with #2033 (or a subsequent supplemental PR) but also get the deterministic land test going. The first part will be a draft PR to supersede #2232 and will be a PR into this branch here. Expect this Monday or Tuesday. I'll code before testing just to open the draft PR for discussion of approach.

**Comment by CoryMartin-NOAA** [comment]
An update, this is taking a bit longer than expected, partially due to other commitments, and partially due to the need for refactoring IMS files to be 'fix' files and not in the DMPDIR. The ball is currently in 2 courts, we need new remapping files generated for the updated fractional grid files, those files placed in the 'correct location', and also script changes are needed, before the test will succeed.

**Comment by aerorahul** [comment]
Thanks for the update.  Please let us know where to get the files from so they can be staged.

**Comment by CoryMartin-NOAA** [comment]
After discussing with @aerorahul I have a minimal PR into this branch to hopefully get the initial test working. Once that is passing and in develop, there will be a few updates coming soon for snow DA, with test changes coming along with them. These will try to be separated into several PRs in order to be more 'agile'. Expect these the week post AMS.

**Comment by aerorahul** [comment]
Thank you @jiaruidong2017 @CoryMartin-NOAA for working on adding a test and improving the reliability of the system

**Pull request merged**
